src/OpenDiffix.Core/Analyzer.fs

```diff
@@ -42,11 +42,19 @@ let private mapFunctionExpression rangeColumns fn parsedArgs =
   let mapAids parsedAids =
     parsedAids |> List.map (mapExpression rangeColumns) |> ListExpr
 
+  let mapAvg aggregateArgs parsedArgs sumFunction countFunction =
+    let sum = mapFunctionExpression rangeColumns (AggregateFunction(sumFunction, aggregateArgs)) parsedArgs
+    let count = mapFunctionExpression rangeColumns (AggregateFunction(countFunction, aggregateArgs)) parsedArgs
+    let castSum = FunctionExpr(ScalarFunction ScalarFunction.Cast, [ sum; Constant(String "real") ])
+    ScalarFunction ScalarFunction.Divide, [ castSum; count ]
+
   (match fn, parsedArgs with
    | AggregateFunction (Count, aggregateArgs), [ ParserTypes.Star ] -> //
      AggregateFunction(Count, aggregateArgs), []
    | AggregateFunction (CountNoise, aggregateArgs), [ ParserTypes.Star ] -> //
      AggregateFunction(CountNoise, aggregateArgs), []
+   | AggregateFunction (Avg, aggregateArgs), parsedArgs -> //
+     mapAvg aggregateArgs parsedArgs Sum Count
```

Comment:
Why do we need this one? Is this the direct access version of avg?

Reply:
I complicated the story. This line is executed for `SELECT avg(age) FROM...` regardless of direct/anonymized access. The other added line here with `DiffixAvg` is executed for `SELECT diffix_avg(age)` in direct access.